### PR TITLE
2675-auth-terms

### PR DESCRIPTION
**Pull request by sahma19**
https://app.zenhub.com/workspaces/team-atlas-633199659e255a37cd1d144f/issues/gh/energinet-datahub/energy-origin-issues/2675

### CODE REVIEW

File: libraries/dotnet/EnergyOrigin.IntegrationEvents/doc/specific-events/authorization-acceptance-of-terms.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Authorization: Acceptance of terms - Integration Event
+
+## Purpose
+
+The OrgAcceptedTerms record is for establishing integration between the Datahub system,
+which stores the organization's metering point data, and the Energy Track & Trace™ domain.
+When an organization accepts the terms and conditions, through the API served by the Authorization subsystem,
+this record is sent to a RabbitMQ message broker and then consumed by the relevant connected systems.
+This is to ensure that the organization's metering point data can be properly accessed, and processed,
```
Comment (tnickelsen) [inline libraries/dotnet/EnergyOrigin.IntegrationEvents/doc/specific-events/authorization-acceptance-of-terms.md:9]: fjern 'properly'

Comment (tnickelsen) [reply libraries/dotnet/EnergyOrigin.IntegrationEvents/doc/specific-events/authorization-acceptance-of-terms.md:9]: Skriv noget om, at der bliver lavet en relation i DH2

File: libraries/dotnet/EnergyOrigin.IntegrationEvents/doc/specific-integration-guides/masstransit-rabbitmq.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+Energy Track & Trace™ currently uses RabbitMQ, as the message broker for publishing and consuming integration events.
+To ensure reliability and consistency between database operations and message publishing,
+[MassTransit's](https://masstransit.io/) transactional outbox pattern is used,
+using the MassTransit.EntityFrameworkCore, and MassTransit.RabbitMQ NuGet packages.
+
+### Setup Guide
+
+To start publishing integration event messages,
+using [Transactional Outbox Configuration](https://masstransit.io/documentation/configuration/middleware/outbox),
```
Comment (tnickelsen) [inline libraries/dotnet/EnergyOrigin.IntegrationEvents/doc/specific-integration-guides/masstransit-rabbitmq.md:13]: Do we need the below text when we have the masstransit guide?

File: libraries/dotnet/EnergyOrigin.IntegrationEvents/README.md [hunk 3]
```diff
@@ -1,5 +1,45 @@
-# Library
+# EnergyOrigin.IntegrationEvents
 
-## EnergyOrigin.IntegrationEvents
+## Overview
 
-The purpose of EnergyOrigin.IntegrationEvents is to have a single library for all the Integration Events that are used in the EnergyOrigin solution.
+The EnergyOrigin.IntegrationEvents NuGet package provides a standardized way to define and use integration events,
```
Comment (tnickelsen) [inline libraries/dotnet/EnergyOrigin.IntegrationEvents/README.md:5]: *Is our way of sharing integration event contracts between our systems

File: libraries/dotnet/EnergyOrigin.IntegrationEvents/README.md [hunk 4]
```diff
@@ -1,5 +1,45 @@
...
+through a message-driven architecture.
+
+## How it works
+
+The package contains an abstract IntegrationEvent record that serves as the base for all integration events.
+This base record includes common properties such as:
+
+- **Id:** A unique identifier for the event.
+- **TraceId:** An identifier for tracing the event through the system.
+- **Created:** The timestamp when the event was created.
```
Comment (tnickelsen) [inline libraries/dotnet/EnergyOrigin.IntegrationEvents/README.md:16]: Har skrevet ADR om integration events her: https://github.com/Energinet-DataHub/energy-origin/blob/main/doc/architecture/adr/0025-integration-events.md

Det er lidt det samme som noget af det her

File: doc/architecture/domains/authorization/features/accept-terms.md [hunk 5]
```diff
@@ -0,0 +1,281 @@
+# Terms Acceptance Documentation
+
+## Overview
+
+The Terms Acceptance API allows organizations to accept the latest terms and conditions.
+When an organization accepts the terms, the Authorization database is updated, and an integration event is published,
+to RabbitMQ, using MassTransit's transactional outbox pattern.
```
Comment (tnickelsen) [inline doc/architecture/domains/authorization/features/accept-terms.md:7]: fjern komma så det bliver 'is published to rabbitMq'

File: doc/architecture/domains/authorization/features/accept-terms.md [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+        DB-->>Handler: Confirm update
+        Handler->>DB: Commit transaction
+        DB-->>Handler: Confirm commit
+        deactivate DB
+    end
+
+    rect rgb(50, 50, 50)
+        note over Handler, RabbitMQ: Publish integration event
+        Handler->>RabbitMQ: Send OrgAcceptedTerms message
+    end
```
Comment (tnickelsen) [inline doc/architecture/domains/authorization/features/accept-terms.md:43]: Why the black boxes?

File: doc/architecture/domains/authorization/features/accept-terms.md [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+  "status": false,
+  "message": "Failed to accept terms."
+}
+```
+
+### Failure (403 Forbidden)
+
+```json
+{}
+```
```
Comment (tnickelsen) [inline doc/architecture/domains/authorization/features/accept-terms.md:103]: Why is this needed? We have a swagger :)

File: doc/architecture/domains/authorization/features/accept-terms.md [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+
+## Implementation Details
```
Comment (tnickelsen) [inline doc/architecture/domains/authorization/features/accept-terms.md:105]: Delete from here on down. The code is documenting this just as well and we don't need a description of the outbox pattern - that is better described elsewhere

File: doc/architecture/domains/authorization/features/authorize-user.md [hunk 9]
```diff
@@ -0,0 +1,210 @@
+﻿# User Consent Authorization Documentation
```
Comment (tnickelsen) [inline doc/architecture/domains/authorization/features/authorize-user.md:1]: Call the folder 'flows' instead of 'features'. We use flows other places

File: doc/architecture/domains/authorization/features/authorize-user.md [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+
+        alt Terms not accepted or outdated
+            Handler->>DB: Update Terms Acceptance Status
+            DB-->>Handler: Confirm Update
+        end
+
+        Handler->>DB: Commit Transaction
+        DB-->>Handler: Confirm Commit
+        deactivate DB
+    end
```
Comment (tnickelsen) [inline doc/architecture/domains/authorization/features/authorize-user.md:64]: Why the black box? Can't see what's inside

File: doc/architecture/domains/authorization/features/authorize-user.md [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+        deactivate DB
+        Handler-->>API: Throw Exception
+        deactivate Handler
+        activate API
+        API-->>B2C: Return Error Response
+        deactivate API
+    end
+```
+
+## Endpoint
```
Comment (tnickelsen) [inline doc/architecture/domains/authorization/features/authorize-user.md:88]: Again, from here on and below can be read in the code. Delete

File: libraries/dotnet/EnergyOrigin.IntegrationEvents/doc/specific-integration-guides/masstransit-rabbitmq.md (hunk 2 above)
Comment (sahma19) [reply libraries/dotnet/EnergyOrigin.IntegrationEvents/doc/specific-integration-guides/masstransit-rabbitmq.md:13]: Godt spørgsmål. Jeg tænker vores rabbitmq options er rimeligt specifik for os, da det er de parametre som eo-base bruger, når den deployer rabbitmq.

File: libraries/dotnet/EnergyOrigin.IntegrationEvents/README.md (hunk 4 above)
Comment (sahma19) [reply libraries/dotnet/EnergyOrigin.IntegrationEvents/README.md:16]: Hvad ønsker du der skal gøres? skal jeg flytte det derhen, eller fjerne det her helt?

File: doc/architecture/domains/authorization/features/accept-terms.md (hunk 6 above)
Comment (sahma19) [reply doc/architecture/domains/authorization/features/accept-terms.md:43]: I think it looks nice. synes du ikke det gør det nemmere at læse?

File: libraries/dotnet/EnergyOrigin.IntegrationEvents/doc/specific-events/authorization-acceptance-of-terms.md (hunk 1 above)
Comment (sahma19) [reply libraries/dotnet/EnergyOrigin.IntegrationEvents/doc/specific-events/authorization-acceptance-of-terms.md]: Done: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/65f35263d8e69117f13905eb7d5a03c5d9bf2698

File: libraries/dotnet/EnergyOrigin.IntegrationEvents/README.md (hunk 3 above)
Comment (sahma19) [reply libraries/dotnet/EnergyOrigin.IntegrationEvents/README.md]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/ecdb7624ce5509d72f389fd47d16a0b32a0fb93e

File: doc/architecture/domains/authorization/features/accept-terms.md (hunk 5 above)
Comment (sahma19) [reply doc/architecture/domains/authorization/features/accept-terms.md:7]: Done: https://github.com/Energinet-DataHub/energy-origin/commit/65f35263d8e69117f13905eb7d5a03c5d9bf2698

Comment (sahma19) [reply doc/architecture/domains/authorization/features/accept-terms.md]: Done: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/b2fbe037256c6e89c711cbe591be89996f85002f

File: doc/architecture/domains/authorization/features/accept-terms.md (hunk 7 above)
Comment (sahma19) [reply doc/architecture/domains/authorization/features/accept-terms.md]: Removed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/0e3afd76de61829946609d649dee4927457e4e2a

File: libraries/dotnet/EnergyOrigin.IntegrationEvents/doc/specific-integration-guides/masstransit-rabbitmq.md (hunk 2 above)
Comment (sahma19) [reply libraries/dotnet/EnergyOrigin.IntegrationEvents/doc/specific-integration-guides/masstransit-rabbitmq.md:13]: Removed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/72e19f5843d53b37751d65c929a13f1f7d24c78f

File: libraries/dotnet/EnergyOrigin.IntegrationEvents/README.md (hunk 4 above)
Comment (sahma19) [reply libraries/dotnet/EnergyOrigin.IntegrationEvents/README.md]: Removed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/26ed9a9c16c5d09abd07adaf2beb24a6a57685b3

File: libraries/dotnet/EnergyOrigin.IntegrationEvents/README.md (hunk 3 above)
Comment (sahma19) [reply libraries/dotnet/EnergyOrigin.IntegrationEvents/README.md]: Fixed for real this time: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/c141531366a64b4d76bddf28fd69773917ee3e33

File: libraries/dotnet/EnergyOrigin.IntegrationEvents/doc/specific-events/authorization-acceptance-of-terms.md (hunk 1 above)
Comment (sahma19) [reply libraries/dotnet/EnergyOrigin.IntegrationEvents/doc/specific-events/authorization-acceptance-of-terms.md]: Done again: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/e39284d7bb70f4a43d68b4ff934e8853b82510fe

File: doc/architecture/domains/authorization/features/authorize-user.md (hunk 10 above)
Comment (sahma19) [reply doc/architecture/domains/authorization/features/authorize-user.md:64]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/5f203707232534e753d4742b5869859da732d5cc

File: doc/architecture/domains/authorization/features/accept-terms.md (hunk 7 above)
Comment (sahma19) [reply doc/architecture/domains/authorization/features/accept-terms.md]: Deleted some more: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/a95f20d81b0d6ddd9af3b7939c59ec9946dc6a73

File: doc/architecture/domains/authorization/features/accept-terms.md (hunk 8 above)
Comment (sahma19) [reply doc/architecture/domains/authorization/features/accept-terms.md]: deleted: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/a95f20d81b0d6ddd9af3b7939c59ec9946dc6a73

File: doc/architecture/domains/authorization/flows/authorize-user.md
Comment (Soped42) [inline doc/architecture/domains/authorization/flows/authorize-user.md:1]: Fix lige grafen tak!

File: domains/authorization/Authorization.API/API/Authorization/[Features]/AcceptTermsCommand.cs [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+                .FirstOrDefaultAsync(o => o.Tin == usersOrganizationsCvr, cancellationToken);
+
+            if (usersAffiliatedOrganization == null)
+            {
+                usersAffiliatedOrganization = Organization.Create(usersOrganizationsCvr, OrganizationName.Create(request.OrgName));
+                await organizationRepository.AddAsync(usersAffiliatedOrganization, cancellationToken);
+            }
+
+            var latestTerms = await termsRepository.Query()
+                .OrderByDescending(t => t.Version)
```
Comment (martinhenningjensen) [inline domains/authorization/Authorization.API/API/Authorization/[Features]/AcceptTermsCommand.cs:44]: Since Version is a string column, this must mean "3.0" > "10.0" ? Is this expected?

File: domains/authorization/Authorization.API/API.UnitTests/API.UnitTests/[Commands]/GetConsentForUserQueryHandlerTests.cs [hunk 13]
```diff
@@ -0,0 +1,138 @@
...
+using API.Models;
+using API.Repository;
+using API.UnitTests.Repository;
+using API.ValueObjects;
+using FluentAssertions;
+using NSubstitute;
+
+namespace API.UnitTests._Commands_;
+
+public class GetConsentForUserQueryHandlerTests
```
Comment (martinhenningjensen) [inline domains/authorization/Authorization.API/API.UnitTests/API.UnitTests/[Commands]/GetConsentForUserQueryHandlerTests.cs:12]: What about the case where a new version of Terms exists, what is the expected behavior in that case?

**Review by martinhenningjensen — CHANGES_REQUESTED** [review]

File: README.md [hunk 14]
```diff
@@ -34,7 +35,13 @@
 
 ## Other Documentation
 
-### Authentication & Authorization
+### Azure B2C
 
 - [Azure B2C Field Guide](https://energinet.atlassian.net/wiki/spaces/EnergyOrigin/pages/902463491/Azure+B2C+Field+Guide) (Requires Confluence access to Energy Origin space)
```
[the remark this answers is not in the document]
Comment (sahma19) [reply README.md:40]: It was created, prior to having specified the "onboarding document" assignment.

File: domains/authorization/Authorization.API/API/Authorization/[Features]/GetConsentForUserCommand.cs [hunk 15]
```diff
@@ -0,0 +1,117 @@
...
+            var orgTin = Tin.Create(command.OrgCvr);
+            var organization = await organizationRepository.Query()
+                .Include(o => o.Affiliations)
+                .FirstOrDefaultAsync(o => o.Tin == orgTin, cancellationToken);
+
+            var latestTerms = await termsRepository.Query()
+                .OrderByDescending(t => t.Version)
+                .FirstOrDefaultAsync(cancellationToken);
+
+            var termsAccepted = false;
```
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/authorization/Authorization.API/API/Authorization/[Features]/GetConsentForUserCommand.cs]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/c2985b74dc1525019483d75c864744fc42f5c4a4

File: domains/authorization/Authorization.API/API/Authorization/[Features]/AcceptTermsCommand.cs [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+                .FirstOrDefaultAsync(cancellationToken);
+
+            if (latestTerms == null)
+            {
+                return false;
+            }
+
+            if (!usersAffiliatedOrganization.TermsAccepted || usersAffiliatedOrganization.TermsVersion != latestTerms.Version)
```
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/authorization/Authorization.API/API/Authorization/[Features]/AcceptTermsCommand.cs:52]: Idk. i would have to pass in a terms object would i not? seems weird

File: domains/authorization/Authorization.API/API/Authorization/Controllers/TermsController.cs [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+
+namespace API.Authorization.Controllers;
+
+[ApiController]
+[ApiVersion(ApiVersions.Version20230101)]
+public class TermsController(IMediator mediator, IdentityDescriptor identityDescriptor) : ControllerBase
+{
+    [HttpPost]
+    [Route("api/authorization/terms/accept")]
+    [Authorize(Policy.B2CCvrClaim)]
```
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/authorization/Authorization.API/API/Authorization/Controllers/TermsController.cs]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/c2985b74dc1525019483d75c864744fc42f5c4a4

File: domains/authorization/Authorization.API/API/Authorization/[Features]/AcceptTermsCommand.cs (hunk 12 above)
Comment (sahma19) [reply domains/authorization/Authorization.API/API/Authorization/[Features]/AcceptTermsCommand.cs:44]: fixed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/e9c31673e661c9e2063fac0775e3e70940d7e0eb

File: domains/authorization/Authorization.API/API.UnitTests/API.UnitTests/[Commands]/GetConsentForUserQueryHandlerTests.cs (hunk 13 above)
Comment (sahma19) [reply domains/authorization/Authorization.API/API.UnitTests/API.UnitTests/[Commands]/GetConsentForUserQueryHandlerTests.cs:12]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/45fba051f7cff543c28fbc209992b389f2ee64ba

File: domains/authorization/Authorization.API/API.UnitTests/API.UnitTests/Controllers/AuthorizationControllerTests.cs
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/authorization/Authorization.API/API.UnitTests/API.UnitTests/Controllers/AuthorizationControllerTests.cs:1]: Removed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/57ce565c19bd604f3d1c9a2b67303325506f5e7c

File: domains/authorization/Authorization.API/API/Authorization/[Features]/AcceptTermsCommand.cs [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+            {
+                usersAffiliatedOrganization = Organization.Create(usersOrganizationsCvr, OrganizationName.Create(request.OrgName));
+                await organizationRepository.AddAsync(usersAffiliatedOrganization, cancellationToken);
+            }
+
+            var latestTerms = await termsRepository.Query()
+                .OrderByDescending(t => t.Version)
+                .FirstOrDefaultAsync(cancellationToken);
+
+            if (latestTerms == null)
```
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/authorization/Authorization.API/API/Authorization/[Features]/AcceptTermsCommand.cs:47]: Yes. Terms need to exist in the database or someone messed up

File: domains/authorization/Authorization.API/API/Authorization/[Features]/AcceptTermsCommand.cs [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+                request.OrgCvr,
+                request.UserId
+            ), cancellationToken);
+
+            await unitOfWork.CommitAsync();
+
+
+            return true;
+        }
+        catch
```
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/authorization/Authorization.API/API/Authorization/[Features]/AcceptTermsCommand.cs]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/c748f20645e390167753c7edac590f1c2e0a95cc

File: domains/authorization/Authorization.API/API/Authorization/[Features]/GetConsentForUserCommand.cs [hunk 20]
```diff
@@ -0,0 +1,117 @@
...
+            if (organization == null || latestTerms == null)
+            {
+                await unitOfWork.RollbackAsync();
+                return new GetConsentForUserCommandResult(
+                    command.Sub,
+                    command.Name,
+                    "User",
+                    command.OrgName,
+                    new List<Guid>(),
+                    "dashboard production meters certificates wallet",
```
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/authorization/Authorization.API/API/Authorization/[Features]/GetConsentForUserCommand.cs]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/c748f20645e390167753c7edac590f1c2e0a95cc

File: domains/authorization/Authorization.API/API/Authorization/[Features]/GetConsentForUserCommand.cs [hunk 21]
```diff
@@ -0,0 +1,117 @@
...
+                    false
+                );
+            }
+
+            termsAccepted = organization.TermsAccepted && organization.TermsVersion == latestTerms.Version;
```
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/authorization/Authorization.API/API/Authorization/[Features]/GetConsentForUserCommand.cs]: Renamed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/a97ce23f8da481761fde9756edfc088baaef31fd

File: domains/authorization/Authorization.API/API/Authorization/Controllers/TermsController.cs [hunk 22]
```diff
@@ -0,0 +1,36 @@
...
+    [ProducesResponseType(typeof(AcceptTermsResponseDto), StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(typeof(AcceptTermsResponseDto), StatusCodes.Status403Forbidden)]
+    public async Task<ActionResult<AcceptTermsResponseDto>> AcceptTerms()
+    {
+        var command = new AcceptTermsCommand(identityDescriptor.OrganizationCvr!, identityDescriptor.OrganizationName, identityDescriptor.Subject);
+        var result = await mediator.Send(command);
+
+        if (!result)
+        {
+            return BadRequest(new AcceptTermsResponseDto(false, "Failed to accept terms."));
```
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/authorization/Authorization.API/API/Authorization/Controllers/TermsController.cs]: Changed to 500 Internal server error: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/627e69251a2f76b16676c96c13e465ffc21fab06

File: domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/Controllers/TermsControllerTests.cs [hunk 23]
```diff
@@ -0,0 +1,97 @@
...
+        var newDatabaseInfo = integrationTestFixture.WebAppFactory.ConnectionString;
+        _options = new DbContextOptionsBuilder<ApplicationDbContext>().UseNpgsql(newDatabaseInfo).Options;
+        _integrationTestFixture = integrationTestFixture;
+    }
+
+    [Fact]
+    public async Task GivenValidRequest_WhenAcceptingTerms_ThenHttpOkAndTermsAccepted()
+    {
+        await using var context = new ApplicationDbContext(_options);
+        await context.Database.EnsureCreatedAsync();
```
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/Controllers/TermsControllerTests.cs]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/2dacb33b8ab30c966b9914b4de14f05c44a376f6

File: domains/authorization/Authorization.API/API/Authorization/Controllers/TermsController.cs [hunk 24]
```diff
@@ -0,0 +1,36 @@
...
+        }
+
+        return Ok(new AcceptTermsResponseDto(true, "Terms accepted successfully."));
+    }
+}
+
+public record AcceptTermsResponseDto(bool Status, string Message);
```
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/authorization/Authorization.API/API/Authorization/Controllers/TermsController.cs]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/b7837d98960e28a326fb207253efcba4f61ab2ef

File: domains/authorization/Authorization.API/API/Authorization/[Features]/GetConsentForUserCommand.cs [hunk 25]
```diff
@@ -0,0 +1,117 @@
...
+
+            if (!termsAccepted)
+            {
+                await unitOfWork.RollbackAsync();
```
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/authorization/Authorization.API/API/Authorization/[Features]/GetConsentForUserCommand.cs]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/98bedee94671319779f2cfde2240f3e61aacbe99

File: doc/architecture/domains/authorization/flows/authorize-user.md [hunk 26]
```diff
@@ -0,0 +1,146 @@
...
+
+## Endpoint
+
+```http request
+POST /api/authorization/user-consent/
+```
+
+## Purpose
+
+The endpoint serves several functions:
```
[the remark this answers is not in the document]
Comment (sahma19) [reply doc/architecture/domains/authorization/flows/authorize-user.md]: fixed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/3bd60b9ed7c68b67ad309571eaad3776b99b44f3

File: doc/architecture/domains/authorization/flows/authorize-user.md [hunk 27]
```diff
@@ -0,0 +1,146 @@
...
+    activate B2C
+    B2C->>API: POST /api/authorization/user-consent/
+    Note over B2C,API: Send user info (sub, name, orgName, orgCvr)
+    activate API
+
+    API->>Handler: Create and send GetConsentForUserCommand
+    Note over API,Handler: Command contains sub, name, orgName, orgCvr
+    activate Handler
+
+    rect rgb(100, 100, 100)
```
[the remark this answers is not in the document]
Comment (sahma19) [reply doc/architecture/domains/authorization/flows/authorize-user.md]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/3bd60b9ed7c68b67ad309571eaad3776b99b44f3

File: doc/architecture/domains/authorization/flows/accept-terms.md [hunk 28]
```diff
@@ -0,0 +1,75 @@
...
+    participant RabbitMQ as RabbitMQ Broker
+
+    User->>API: POST /api/authorization/terms/accept
+    activate API
+    API->>API: Validate JWT and extract claims
+    API->>Handler: Send AcceptTermsCommand
+    activate Handler
+
+    rect rgb(100, 100, 100)
+        Note over Handler, DB: Database Interactions
```
[the remark this answers is not in the document]
Comment (sahma19) [reply doc/architecture/domains/authorization/flows/accept-terms.md]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/3bd60b9ed7c68b67ad309571eaad3776b99b44f3

File: doc/architecture/domains/authorization/authorization.md
[the remark this answers is not in the document]
Comment (sahma19) [reply doc/architecture/domains/authorization/authorization.md:1]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/a5afbfc1f91593558ea9fc8af16f45e91ebe53cd

File: README.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (sahma19) [reply README.md]: It is now removed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/a5afbfc1f91593558ea9fc8af16f45e91ebe53cd

File: doc/architecture/domains/authorization/authorization.md
[the remark this answers is not in the document]
Comment (sahma19) [reply doc/architecture/domains/authorization/authorization.md:1]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/3bd60b9ed7c68b67ad309571eaad3776b99b44f3

File: doc/architecture/domains/authorization/flows/accept-terms.md
[the remark this answers is not in the document]
Comment (sahma19) [reply doc/architecture/domains/authorization/flows/accept-terms.md:1]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/3bd60b9ed7c68b67ad309571eaad3776b99b44f3

File: doc/architecture/domains/authorization/flows/authorize-user.md
[the remark this answers is not in the document]
Comment (sahma19) [reply doc/architecture/domains/authorization/flows/authorize-user.md:1]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/3bd60b9ed7c68b67ad309571eaad3776b99b44f3

File: doc/architecture/domains/authorization/flows/authorize-user.md [hunk 29]
```diff
@@ -0,0 +1,146 @@
...
+4. Manages user-organization affiliations.
+5. Provides authorization information for the user.
+
+### Azure B2C Integration
+
+This endpoint is designed to be called exclusively by the Azure B2C tenant as part of the authentication flow.
+The B2C tenant sends user information obtained during the sign-in process,
+and this endpoint responds with the necessary authorization details.
+
+## Possible Scenarios
```
[the remark this answers is not in the document]
Comment (sahma19) [reply doc/architecture/domains/authorization/flows/authorize-user.md]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/3bd60b9ed7c68b67ad309571eaad3776b99b44f3

**Review by martinhenningjensen — APPROVED** [review]
